src/main/java/org/springframework/cloud/deployer/spi/kubernetes/DefaultContainerFactory.java

```diff
@@ -273,9 +273,16 @@ private ProbeType getProbeType(ContainerConfiguration containerConfiguration) {
 	 */
 	List<String> createCommandArgs(AppDeploymentRequest request) {
 		List<String> cmdArgs = new LinkedList<>();
-		List<String> commandArgOptions = request.getCommandlineArguments().stream()
-				.map(arg-> arg.substring(0,arg.indexOf("=")).replaceAll("^--",""))
-				.collect(Collectors.toList());
+
+		List<String> commandArgOptions = new ArrayList<>();
+		for(String arg : request.getCommandlineArguments()) {
+			int indexOfAssignment = arg.indexOf("=");
+			if(indexOfAssignment < 0) {
+				commandArgOptions.add(arg);
+			} else {
+				commandArgOptions.add(arg.substring(0, indexOfAssignment).replaceAll("^--",""));
+			}
+		}
```

Comment:
Should we also scrub the arg for extra whitespace (trim) so that it matches up against overrides from app definition?

Reply:
For now I'd say no.   If there is a need we can certainly strip it later.

Reply:
The current commit is actually doing the `trim()` as it was in my suggested code for that method. Just making sure you are ok w/ that.

Reply:
Ahh.  gotcha.   
I went back and forth.  I guess I said yes by using the code you suggested.